config/tests/src/test/java/org/jboss/windup/config/TestInsiderRuleProvider.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.windup.config;
+
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.graph.GraphContext;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ *
+ * @author Ondrej Zizka, ozizka at redhat.com
+ */
+public abstract class TestInsiderRuleProvider extends WindupRuleProvider implements Insider {
```

Comment:
Actually, I think a simpler way is just to leave GraphCondition and GraphOperation, and but rather re-implement the functionality of each within this class. Then you don't need to worry about new interfaces or refactoring.